lucene/core/src/java/org/apache/lucene/codecs/VectorValues.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.codecs;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.util.BytesRef;
+
+/**
+ * Holds the vector values for a particular field.
+ *
+ * @lucene.experimental
+ */
+public abstract class VectorValues {
+
+  public abstract DocIdSetIterator iterator();
+
+  public abstract int docID();
```

Comment:
It will return the doc ID that the iterator is currently positioned on. I took inspiration from the `Scorer` interface here. Although looking at it again, I can definitely see how this could be confusing.

Reply:
I tend to agree with @msokolov that we might want to follow the Numeric/BinaryDocValues examples instead, which extend DocIdSetIterator.